src/OpenTelemetry.AutoInstrumentation/Instrumentation.cs

```diff
@@ -87,20 +91,18 @@ public static void Initialize()
 
 #if NET6_0_OR_GREATER
             var profilerEnabled = GeneralSettings.Value.ProfilerEnabled;
-            var threadSamplingEnabled = false;
-            var allocationSamplingEnabled = false;
-            TimeSpan exportInterval = default;
-            object? continuousProfilerExporter = null;
 
             if (profilerEnabled)
             {
-                (threadSamplingEnabled, var threadSamplingInterval, allocationSamplingEnabled, var maxMemorySamplesPerMinute, exportInterval, continuousProfilerExporter) = _pluginManager.GetFirstContinuousConfiguration();
+                var (threadSamplingEnabled, threadSamplingInterval, allocationSamplingEnabled, maxMemorySamplesPerMinute, exportInterval, continuousProfilerExporter) = _pluginManager.GetFirstContinuousConfiguration();
                 Logger.Debug($"Continuous profiling configuration: Thread sampling enabled: {threadSamplingEnabled}, thread sampling interval: {threadSamplingInterval}, allocation sampling enabled: {allocationSamplingEnabled}, max memory samples per minute: {maxMemorySamplesPerMinute}, export interval: {exportInterval}, continuous profiler exporter: {continuousProfilerExporter.GetType()}");
 
                 if (threadSamplingEnabled || allocationSamplingEnabled)
                 {
                     NativeMethods.ConfigureNativeContinuousProfiler(threadSamplingEnabled, threadSamplingInterval, allocationSamplingEnabled, maxMemorySamplesPerMinute);
-                    Activity.CurrentChanged += ContinuousProfilerProcessor.Activity_CurrentChanged;
+
+                    _profilerProcessor = new ContinuousProfilerProcessor(threadSamplingEnabled, allocationSamplingEnabled, exportInterval, continuousProfilerExporter);
+                    Activity.CurrentChanged += _profilerProcessor.Activity_CurrentChanged;
```

Comment:
it may create more activities if `_profilerProcessor` is disposed, so could trigger nullreference exception?

Reply:
changed in https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/3294/commits/d6db263c07d68c664dfa91fa46118c3deb375827